tools/functional-tester/etcd-tester/failpoint.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+// fpRetries is number of times to retry in case etcd isn't fully booted
+const fpRetries = 5
```

Comment:
The agent start rpc returns as soon as the agent gets etcd's pid, so it's possible the tester will try to read off the failpoints before etcd sets up the failpoint server. I see the code should probably use `WaitHealth` before reading off the failpoints so that there's no need to retry.